File: Frontend/MovieBookingAppUi/src/app/ticket/ticket.ts
import { ShowTime } from "../showTime/show-time";

export class Ticket {
    constructor(public ticketId:number,
        public showTime:ShowTime,
        public userName:string,
        public seatsBooked:string[],
        public totalPrice:number,
        public ticketBookedDateTime:string){}
}

File: Backend/Movie/src/main/java/com/cts/movie/model/Ticket.java
package com.cts.movie.model;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Ticket {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int ticketId;

	@ManyToOne(fetch = FetchType.EAGER)
	@JoinColumn(name = "showTime")
	private ShowTime showTime;

	private String userName;

	private String[] seatsBooked;

	private int totalPrice;

	private String ticketBookedDateTime;

}
